### PR TITLE
Private workflows fix

### PR DESCRIPTION
**Pull request by jsiever1**
Fix to skip the workflows in a Private repository. Only workflows from a public or internal repo will be included.

### CODE REVIEW

**Comment by jsiever1** [comment]
I can make it optional.

**Comment by jsiever1** [comment]
Made scan of workflows optional with default true and also made the scan for private workflows optional with default false

File: src/main.ts [hunk 1]
```diff
@@ -110,15 +115,12 @@ export class ActionContent {
...
-  //owner: string | undefined
   repo: string | undefined
   downloadUrl: string | undefined
-  //author: string | undefined
   description: string | undefined
   forkedfrom: string | undefined
-  //readme: string | undefined
-  //using: string | undefined
   isArchived: boolean | undefined
+  visability: string | undefined
```
Comment (rajbos) [inline src/main.ts:123]: Typo
```suggestion
  visibility: string | undefined
```

File: src/main.ts [hunk 2]
```diff
@@ -693,19 +695,25 @@ async function getAllReusableWorkflowsUsingSearch(
     const repoName = searchResult[index].repository.name
     const repoOwner = searchResult[index].repository.owner.login
 
-    // Push workflow to the list
-    core.info(`Found workflow ${fileName } in ${repoName}/${filePath}`)
-
     // Get the Repository Details
     const repoDetail = await getRepoDetails(client, repoOwner, repoName)
     const isArchived = repoDetail.archived
+    const visability = repoDetail.visibility
```
Comment (rajbos) [inline src/main.ts:701]:
```suggestion
    const visibility = repoDetail.visibility
```

File: src/main.ts [hunk 3]
```diff
@@ -693,19 +695,25 @@ async function getAllReusableWorkflowsUsingSearch(
...
+
+    // Skip workflow if it is a private repo
+    if ( includePrivateWorkflows === 'false' && visability === 'private') {
```
Comment (rajbos) [inline src/main.ts:704]:
```suggestion
    if ( includePrivateWorkflows === 'false' && visibility === 'private') {
```

File: src/main.ts [hunk 4]
```diff
@@ -693,19 +695,25 @@ async function getAllReusableWorkflowsUsingSearch(
...
+    core.info(`Found workflow ${fileName } in ${repoName}/${filePath}`)
 
     const result = await getWorkflowInfo(
       client,
       repoOwner,
       repoName,
       filePath,
-      isArchived
+      isArchived,
+      visability
```
Comment (rajbos) [inline src/main.ts:716]:
```suggestion
      visibility
```

File: src/main.ts [hunk 5]
```diff
@@ -722,7 +730,8 @@ async function getWorkflowInfo(
   owner: string,
   repo: string,
   path: string,
-  isArchived: boolean = false
+  isArchived: boolean = false,
+  visability: string
```
Comment (rajbos) [inline src/main.ts:734]:
```suggestion
  visibility: string
```

File: src/main.ts [hunk 6]
```diff
@@ -746,9 +755,9 @@ async function getWorkflowInfo(
     result.name = yaml.name.replace('.yml', '')
   }
 
-  //result.name = workflowYaml.name
   result.repo = repo
   result.isArchived = isArchived
+  result.visability = visability
```
Comment (rajbos) [inline src/main.ts:760]:
```suggestion
  result.visibility = visibility
```

**Pull request merged**